src/Compressors/sampling.jl

```diff
@@ -0,0 +1,225 @@
+"""
+    Sampling <: Compressor
+
+An implementation of the sampling compression method. This method subsets the rows 
+or columns of a matrix according to a user-supplied distribution. The size of the 
+subset is also provided by the user.
+
+# Mathematical Description
+
+Let ``A`` be an ``m \\times n`` matrix that we want to compress.
+
+If we want to compress ``A`` from the left (i.e., we reduce the number of rows), then
+we create an index set to contain all the indices of selected rows. The indices are 
+chosen by sampling over all the rows with the user-specified distribution in the 
+`distribution` field..
+
+If ``A`` is compressed from the right (i.e., we reduce the number of columns), then
+we create an index set to contain all the indices of selected columns. The indices 
+are chosen by sampling over all the columns with the user-specified distribution 
+in the `distribution` field.
+
+# Fields
+- `cardinality::Cardinality`, the direction the compression matrix is intended to be
+    applied to a target matrix or operator. Values allowed are `Left()` or `Right()`.
+- `compression_dim::Int64`, the target compression dimension. Referred to as ``s`` in 
+    the mathematical description.
+- `distribution::Distribution`, the distribution being used to assign probability weights
+    on the indices.
+
+# Constructor
+
+    Sampling(;cardinality = Left(), compression_dim = 2, distribution)
+
+## Arguments
+- `cardinality::Cardinality`, the direction the compression matrix is intended to be
+    applied to a target matrix or operator. Values allowed are `Left()` or `Right()`.
+    By default `Left()` is chosen.
+- `compression_dim::Int64`, the target compression dimension. Referred to as ``s`` in the
+    mathemtical description. By default this is set to 2.
+- `distribution::Distribution`, the distribution being used to assign probability weights
+    on the indices. By default this is set as `Uniform` distribution.
+
+## Returns
+- A `Sampling` object.
+
+## Throws
+- `ArgumentError` if `compression_dim` is non-positive
+- `ArgumentError` if `Undef()` is taken as the input for `cardinality`
+"""
+struct Sampling <: Compressor
+    cardinality::Cardinality
+    compression_dim::Int64
+    distribution::Distribution # Function that returns a probability vector over indices
+    function Sampling(cardinality, compression_dim, distribution)
+        # the compression dimension must be positive
+        if compression_dim <= 0
+            throw(ArgumentError("Field `compression_dim` must be positive."))
+        end
+
+        if cardinality == Undef()
+            throw(ArgumentError("`cardinality` must be specified as `Left()` or `Right()`.\
+            `Undef()` is not allowed in `CountSketch` structure."))
```

Comment:
This is not the `CountSketch` structure. When copying information from another file, you need to check it. I will fix this.